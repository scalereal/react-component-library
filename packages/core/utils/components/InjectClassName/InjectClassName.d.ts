import * as React from 'react';
import { Props } from './types';
declare const InjectClassName: React.FunctionComponent<Props>;
export default InjectClassName;
