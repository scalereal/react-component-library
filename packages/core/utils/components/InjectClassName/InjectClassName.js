import React from 'react';
var InjectClassName = function (props) {
    var className = props.className, children = props.children;
    return (React.createElement(React.Fragment, null, React.Children.map(children, function (child) {
        return React.cloneElement(child, {
            className: className
        });
    })));
};
export default InjectClassName;
