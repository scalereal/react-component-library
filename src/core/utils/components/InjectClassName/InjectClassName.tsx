import * as React from 'react';
import { Props } from './types';

const InjectClassName: React.FunctionComponent<Props> = props => {
    const { className, children } = props;
    return (
        <React.Fragment>
            {React.Children.map(children, child => {
                return React.cloneElement(child as any, {
                    className
                });
            })}
        </React.Fragment>
    );
};

export default InjectClassName;
