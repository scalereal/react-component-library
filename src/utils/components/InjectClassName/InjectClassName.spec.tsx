import { render } from '@testing-library/react'
import React from 'react'
import InjectClassName from './InjectClassName'

describe('InjectClassName', () => {
    it('should inject className prop to all children', () => {
        const { container } = render(
            <InjectClassName className="test-className">
                <div>Test div 1</div>
                <div>Test div 2</div>
            </InjectClassName>
        );
        expect(container).toMatchSnapshot();
    });
});
