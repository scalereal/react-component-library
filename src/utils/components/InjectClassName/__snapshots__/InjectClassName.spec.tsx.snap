// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InjectClassName should inject className prop to all children 1`] = `
<div>
  <div
    class="test-className"
  >
    Test div 1
  </div>
  <div
    class="test-className"
  >
    Test div 2
  </div>
</div>
`;
